src/atom_finder/classifier/implicit-predicate.clj

```diff
@@ -0,0 +1,38 @@
+(in-ns 'atom-finder.classifier)
+(import '(org.eclipse.cdt.core.dom.ast  IASTIfStatement IASTForStatement IASTWhileStatement IASTDoStatement IASTBinaryExpression))
+
+(def logical-operators #{IASTBinaryExpression/op_equals
+                        IASTBinaryExpression/op_greaterEqual
+                        IASTBinaryExpression/op_greaterThan
+                        IASTBinaryExpression/op_lessThan
+                        IASTBinaryExpression/op_lessEqual
+                        IASTBinaryExpression/op_logicalAnd
+                        IASTBinaryExpression/op_logicalOr
+                        IASTBinaryExpression/op_notequals})
+
+(defn implicit-preciate-for-if-statement?
+  "Is there an implicit predicate in this if or for statement"
+  [node]
+  (cond
+    (instance? IASTBinaryExpression (.getConditionExpression node))
+      (not (contains? logical-operators (.getOperator (.getConditionExpression node))))
+    :else true))
+
+(defn implicit-predicate-while-do-loop?
+  "Is there an implicit predicate in this do-while loop"
+  [node]
+  (cond
+    (instance? IASTBinaryExpression (.getCondition node))
+      (not (contains? logical-operators (.getOperator (.getCondition node))))
+    :else true))

```

Comment:
`implicit-predicate-while-do-loop?` and `implicit-preciate-for-if-statement` are basically identical, and they both throw exceptions if you pass in the wrong type. I think you can fix both issues by moving the `.getConditionExpression`/`.getCondition` outside the function into the calling code